### DISCUSSION TITLE
Sourcemaps for vscode not found after turbopack

### DISCUSSION THREAD

**Discussion by jandnil** [vercel/turborepo#5926] [Help]
Hi! I'm experiencing a problem:

After applying Turbopack to my project, the code that requires debugging is now mapped to the "turbopack/[project]/folders" location when I inspect it in the browser's developer tools. However, I'm encountering an issue where breakpoints do not seem to work within Visual Studio Code (VSCode). I suspect this problem is related to how the sourcemaps are resolved in the launch.json configuration file. While breakpoints function as expected in the browser's developer tools, they are either not binding or being ignored in VSCode. When I inspect the loaded sourcemaps in VSCode, it appears that the mappings differ from those recognized by the browser's developer tools. What changes should I make to the launch.json file in order to ensure that VSCode correctly resolves the sourcemaps and allows breakpoints to function properly?

In vscode the loaded scripts resolves to:

url: file:////turbopack/%5Bproject%5D/apps/web/pages/index.tsx
absolutePath: /turbopack/[project]/apps/web/pages/index.tsx
absolutePath verified?
✅ From sourcemap, assumed correct
sourcemap children:
None (does not have a sourcemap)
referenced from sourcemap:
%5Bproject%5D__fe8a2a._.js (#847351582) as /turbopack/[project]/apps/web/pages/index.tsx → this
 
Example of simple launch.json  used
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Launch Edge",
      "request": "launch",
      "type": "msedge",
      "url": "http://localhost:3000",   
      "webRoot": "${workspaceFolder}",
      "sourceMaps": true,
        }
  ]
}

**Comment by jandnil** [comment]
Anyone have an idea / input on this?

**Comment by jandnil** [comment]
Still not anyone having some ideas or suggestions ... ?

**Comment by jandnil** [comment]
Anyone have an idea here ?

**Comment by Skarvion** [comment]
Hello @jandnil , I just fixed the issue today. I've posted my comment in NextJS repo [here](https://github.com/vercel/next.js/issues/56702#issuecomment-1913443304)

Hope that helps
